Running code-format for analysis-reconstruction

cmsbuild opened the pull request:
Applying code-format for CMSSW category analysis,reconstruction.
See the build logs here https://cmssdt.cern.ch/jenkins/job/GitHub-refactor-cmssw-module/365//console

cms-bot has successfully run the following:
- scram build code-checks-all
- scram build code-format-all
- scram build

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27115/10225

- This PR adds an extra 12KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @cmsbuild  for master.

It involves the following packages:

CommonTools/CandAlgos
JetMETCorrections/Type1MET
PhysicsTools/PatUtils
PhysicsTools/SelectorUtils


@perrotta, @cmsbuild, @santocch, @slava77 can you please review it and eventually sign? Thanks.
@TaiSakuma, @gouskos, @rappoccio, @HeinerTholen, @ahinzmann, @peruzzim, @seemasharmafnal, @mmarionncern, @makortel, @smoortga, @acaudron, @jdolen, @drkovalskyi, @ferencek, @jdamgov, @nhanvtran, @gkasieczka, @schoef, @clelange, @JyothsnaKomaragiri, @mverzett, @gpetruc, @mariadalfonso, @pvmulder this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-1

Tested at: 4bb99cd4a7d5ad7c9e171d8464bf9d76ea44e51a

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-97757d/777/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation warning when building:  See details on the summary page.

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

perrotta commented on PhysicsTools/SelectorUtils/interface/CutApplicatorBase.h:
```diff
@@ -33,72 +33,66 @@ namespace reco {
-  
-  result_type operator()(const argument_type&) const override
+
+  result_type operator()(const argument_type&) const
```
Why ```override``` got lost?
Please fix

smuzaffar replied:
clang-tidy removed it.

```
/pool/condor/dir_9015/jenkins/workspace/GitHub-refactor-cmssw-module/CMSSW_11_0_X_2019-06-04-2300/src/PhysicsTools/SelectorUtils/interface/CutApplicatorBase.h:62:15: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
  result_type operator()(const argument_type&) const override
              ^                                      ~~~~~~~~
Suppressed 1984 warnings (1975 in non-user code, 8 NOLINT, 1 with check filters)
```
I will check why clang compiler did not see the `final` keyword

smuzaffar committed:
llvm warning fix while building rot dicts

clang-tidy removes the `override` keyword as operator is already marked `final` but during ROOT dictionary build `final` is hidden due to 
`#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)` condition and generate warning. 

```
PhysicsTools/SelectorUtils/interface/CutApplicatorBase.h:62:15: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
  result_type operator()(const argument_type&) const override
              ^                                      ~~~~~~~~
Suppressed 1984 warnings (1975 in non-user code, 8 NOLINT, 1 with check filters)
```

smuzaffar commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27115/10309

- This PR adds an extra 12KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #27115 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

cmsbuild commented:
+1
Tested at: 82d8be6a067f8ab4e8f656105ba7a2a299811417
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-97757d/865/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-97757d/865/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3215458
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215123
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

perrotta commented:
+1
- Technical
- Jenkins tests pass

fabiocos commented:
@santocch now the PR looks ok to me, please check, I will move forward with its integration

santocch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
+1
